diff -up grads-2.0.2/m4/udunits.m4.udunits2 grads-2.0.2/m4/udunits.m4
--- grads-2.0.2/m4/udunits.m4.udunits2	2008-08-06 06:39:53.000000000 -0600
+++ grads-2.0.2/m4/udunits.m4	2014-08-11 15:13:20.084294502 -0600
@@ -5,9 +5,9 @@ AC_DEFUN([GA_CHECK_UDUNITS],
 [
   ga_use_udunits='no'
   AC_CHECK_HEADER([udunits.h],
-  [  AC_CHECK_LIB([udunits],[utInit],
+  [  AC_CHECK_LIB([udunits2],[utInit],
      [  ga_use_udunits='yes'
-        UDUNITS_LIBS='-ludunits'
+        UDUNITS_LIBS='-ludunits2'
      ])
   ])
   
